### PR TITLE
Some XRSpace improvements

### PR DESCRIPTION
**Pull request by Manishearth**
Proper XRSpace support is blocked on https://github.com/immersive-web/webxr/issues/565 , but in the meantime this improves XRSpace support a little bit, preparing both for support in getViewerPose and getPose as well as handling input spaces eventually.

r? @jdm

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23055)

### CODE REVIEW

File: components/script/dom/xrframe.rs [hunk 1]
```diff
@@ -52,16 +54,17 @@ impl XRFrameMethods for XRFrame {
...
-        // We assume the reference space is eye level for now
-        // since it's the only one 3DOF devices support
-        if reference.is_some() {
-            // it's not possible to obtain a reference
-            // space at all yet
-            return None;
+    fn GetViewerPose(&self, reference: &XRReferenceSpace) -> Option<DomRoot<XRViewerPose>> {
+        if let Some(_) = reference.downcast::<XRStationaryReferenceSpace>() {
+            // For 3DOF devices all three kinds of reference spaces are identical
+            // XXXManishearth support originOffset
```
Comment (asajeffrey) [inline components/script/dom/xrframe.rs]: If we're checking this in, it should be a TODO preferably with a link to an issue.

File: components/script/dom/xrframe.rs [hunk 2]
```diff
@@ -52,16 +54,17 @@ impl XRFrameMethods for XRFrame {
...
+            let left = XRView::new(&self.global(), &self.session, XREye::Left, &self.data);
+            let right = XRView::new(&self.global(), &self.session, XREye::Right, &self.data);
+            Some(XRViewerPose::new(&self.global(), &left, &right))
+        } else {
+            // XXXManishearth support identity reference spaces
```
Comment (asajeffrey) [inline components/script/dom/xrframe.rs]: Ditto.

**Review by asajeffrey — APPROVED** [review]
LGTM, with some commenting nits. What's our testing story? Should we add some WPT tests?

**Comment by Manishearth** [comment]
The spec is in a state of flux so there aren't any tests IIRC :( Mostly trying to keep up with it and build scaffolding for features which aren't yet finalized (but I know the general shape of)

**Comment by Manishearth** [comment]
@bors-servo r=asajeffrey

Filed https://github.com/servo/servo/issues/23070

I actually figured out the spec issue behind the fixme (see https://github.com/immersive-web/webxr/issues/565#issuecomment-475086315), but I'll let this land first. 

Filed testing issue https://github.com/servo/servo/issues/23071

**Pull request merged**